src/openrct2-ui/windows/EditorParkEntrance.cpp

```diff
@@ -0,0 +1,380 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2024 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#include <openrct2-ui/interface/Viewport.h>
+#include <openrct2-ui/interface/ViewportInteraction.h>
+#include <openrct2-ui/interface/Widget.h>
+#include <openrct2-ui/windows/Window.h>
+#include <openrct2/Context.h>
+#include <openrct2/Input.h>
+#include <openrct2/actions/ParkEntrancePlaceAction.h>
+#include <openrct2/audio/audio.h>
+#include <openrct2/object/EntranceObject.h>
+#include <openrct2/object/ObjectManager.h>
+#include <openrct2/paint/tile_element/Paint.TileElement.h>
+#include <openrct2/sprites.h>
+#include <openrct2/world/tile_element/Slope.h>
+
+namespace OpenRCT2::Ui::Windows
+{
+    static constexpr StringId kWindowTitle = STR_OBJECT_SELECTION_PARK_ENTRANCE;
+    static constexpr int32_t kImageSize = 116;
+    static constexpr int32_t kNumColumns = 4;
+    static constexpr int32_t kNumRows = 1;
+    static constexpr int32_t kScrollPadding = 2;
+    static constexpr int32_t kScrollWidth = (kImageSize * kNumColumns) + kScrollBarWidth + 4;
+    static constexpr int32_t kScrollHeight = (kImageSize * kNumRows);
+    static constexpr int32_t kWindowWidth = kScrollWidth + 28;
+    static constexpr int32_t kWindowHeight = kScrollHeight + 50;
+
+    struct EntranceSelection
+    {
+        ObjectEntryIndex entryIndex = OBJECT_ENTRY_INDEX_NULL;
+        StringId stringId = STR_NONE;
+        ImageIndex imageId = static_cast<ImageIndex>(SPR_NONE);
```

Comment:
`SPR_NONE` is declared as a signed integer, whereas `ImageIndex` is unsigned. I think we should make this an unsigned throughout, but doing so is definitely beyond the scope of this PR.